### ISSUE TITLE
Enhancement: new events

### ISSUE THREAD

**Issue by morlanius** [PC-Logix/LanteaCraft#272]
It would be nice to have inboundTraveler, outboundTraveler events, they should have values for the name of the player

**Comment by AfterLifeLochie** [comment]
My understanding was that there were already events of this sort, but they didn't say who or what generated the events. @MichiyoRavencroft would have to confirm this, though.

**Comment by MichiyoRavencroft** [comment]
There are not any events for this, I didn't know of any way to raise an event for that.

**Comment by AfterLifeLochie** [comment]
OK. I'll add `entityOutgoing` and `entityIncoming` as events, with the following conditions:
* You won't know what the entity actually is.
* You won't receive an `entityIncoming` event if the entity collides with an iris.

**Comment by MichiyoRavencroft** [comment]
Sounds good to me.

**Comment by hunator** [comment]
@AfterLifeLochie Could you also add some timer that after entityOutgoing gate closes if doesn't recieve another (as in series)? Configurable within config file defalt let says in range 5-15s?
Atherwise it should stay opened for time already set in config or to manual shutdown.

Maybe with exception of active transmission in future (if it will work as relay for OC/CC).

**Comment by morlanius** [comment]
sounds great!

**Comment by morlanius** [comment]
@hunator thats why theres an API dude, some basic dialing computer should handle that, but until they set up that gates in a connected state are basicly chunk loaders then it would need a chunkloader for the OC/CC device to operate it after you leave the area.

**Comment by hunator** [comment]
Yes, I know that timing could be done by dialing computer, but it doesn't influence behaviour of DHD itself. This way, you have to use OC/CC to close gate automaticaly or to manualy on either end (in series gate could be closed only from gate which initiated connection) or you have to run out of maximum time that gate could support connection (in series 38 mins).

If there will be timer for that, it will behave like in show. So gate with DHD only would close after few secs after the last of entities went through.

The OC/CC relay was some idea, that I read here somewere and I think if they impelent that, they also should reset such timer if a message is transported through such relay.

**Comment by morlanius** [comment]
if i select the "auto" option in my dialing software it will close automatically after a timed period the connection is made.

**Comment by hunator** [comment]
@morlanius You are talking about dialing SW -> dialing computer is needed. I'm talking about DHD without dialing computer.
If it should have 100% cannonical behaviour, DHD itself should close a gate few seconds after something goes through. If you watched any SG series, you can see that gates won't stay opened for whole 38mins every time. For SGC it is clear - thay can close it manualy whenever thay want (in those 38mins), but gates with DHD closes automaticaly and there is no way in series how to close it manualy (other then cutting power).
There was just way how to prolong connection duration with DHD - that something was going through (entities, radio transmission, beams, grav. field etc.) or something was stucked in ither gate.
And only strong grav. field (singularity) was able to extend time of connection to more than 38mins and only if singularity was on receiver site.

So what I'm proposing is that DHD (not OC/CC) would save energy by default the same way like in series. So every entityOugoing event would reset timer to some secondsToCloseByEntityPassing.

That means:
- If somethings going through gate should stay open.
- If it is not - it should close in user defined time (few seconds).
- Maximum time of connection should stay secondsToStayOpen (in series 38min).
- There should be two user defined times in config time.

**Comment by morlanius** [comment]
I see what you are saying. Then the stargate.disconect() method albeit handy, is not true to the series also? I don't remember if their dialing computer could close the gate, I remember them able to hold the connection open with Jack's arm or a radio carrier. Perhaps to maintain accuracy then the disconnect function should not work when the connection is inbound?

**Comment by morlanius** [comment]
damn it, edit button doesn't work. punching the red button on a dhd closes the gate.

**Comment by hunator** [comment]
@morlanius I'm saying that only Outgoing gate can control duration of connection and that I have never seen that gate with DHD was ever closed manualy by that DHD (by pressing central button once again).
So yes inbound gate shouldn't be able to close connection (but I understand that it is needed now - but it won't be needed if/after there will be that timer I'm proposing and iris/shield). Now it is handy because it is the only way how to block incomming players and it also doesn't block incoming gate by connection that is not needed anymore (but that was never in series, in fact SGC gate was blocked by incomming connection several times).

**Comment by AfterLifeLochie** [comment]
I have locked this issue from further discussion. *Do not* derail suggestion issues with other bugs or suggestions.

**Comment by AfterLifeLochie** [comment]
This can be done in the current code base with minimal impact or work really.
